### ISSUE TITLE
使用yelp2022在case_study_example.py中報錯

### ISSUE THREAD

**Issue by emilyjeng** [RUCAIBox/RecBole#1531]
您好，我使用了官方提供yelp2022的dataset在case_study_example.py運行，會跑出ValueError: Neither [dataset/yelp2022] exists in the device nor [yelp2022] a known dataset name.，但是我已在RecBole/dataset中放入yelp2018。

以下是我的代碼

import torch
from recbole.utils.case_study import full_sort_topk, full_sort_scores
from recbole.quick_start import load_data_and_model

if __name__ == "__main__":
    config, model, dataset, train_data, valid_data, test_data = load_data_and_model(
        model_file="../saved/SRGNN-Nov-02-2022_04-12-45.pth",
    )  # Here you can replace it by your model path.

#     uid_series = np.array([1, 2])  # internal user id series
    # or you can use dataset.token2id to transfer external user token to internal user id
    #uid_series = dataset.token2id(dataset.uid_field, ["196", "186"])
    uid_series = dataset.token2id(dataset.uid_field, ["bv2nCi5Qv5vroFiqKGopiw", "bv2nCi5Qv5vroFiqKGopiw"])

    topk_score, topk_iid_list = full_sort_topk(
        uid_series, model, test_data, k=10, device=config["device"]
    )
    print(topk_score)  # scores of top 10 items
    print(topk_iid_list)  # internal id of top 10 items
    external_item_list = dataset.id2token(dataset.iid_field, topk_iid_list.cpu())
    print(external_item_list)  # external tokens of top 10 items
    print()

    score = full_sort_scores(uid_series, model, test_data, device=config["device"])
    print(score)  # score of all items
    print(
        score[0, dataset.token2id(dataset.iid_field, ["0W4lkclzZThpx3V65bVgig", "AEx2SYEUJmTxVVB18LlCwA"])]

**Comment by emilyjeng** [comment]
謝謝回應，我確實放在RecBole/dataset/yelp2022中
<img width="143" alt="image" src="https://user-images.githubusercontent.com/74802442/201252815-ba060c77-b883-427d-a1d8-37289752508f.png">

**Comment by emilyjeng** [comment]
您好，重新放置路徑與檔案後已可運行。
想在請問一個問題，我參考#1389，將context_aware，的模型切换评价指标从评分改为排序，並使用yelp2022數據集，但預測出來的結果都非常低，不知道是否我有設置錯誤的地方，以下是我的執行指令python run_recbole.py --model=DeepFM --dataset=yelp2022 --config_files=test.yaml --params_file=hyper.test

--------------------------------------------------------------------------------------------------------------------------------
#test.yaml

field_separator: "\t"
seq_separator: " "
USER_ID_FIELD: user_id
ITEM_ID_FIELD: item_id
RATING_FIELD: rating
TIME_FIELD: timestamp
NEG_PREFIX: neg_
LABEL_FIELD: label
threshold:
    rating: 4
load_col:
    inter: ['user_id', 'item_id', 'rating', 'timestamp']
    user: ['user_id', 'user_name', 'user_useful'] 
    item: ['item_id', 'item_name','city', 'item_stars', 'categories']

# model config
embedding_size: 64               # (int) The embedding size of features.
mlp_hidden_size: [128, 128, 128]  # (list of int) The hidden size of MLP layers.  
dropout_prob: 0.2                 # (float) The dropout rate.
normalize_all: True #正規化

epochs: 500
train_batch_size: 4096
eval_batch_size: 40960000
metrics: ['Recall', 'MRR', 'NDCG', 'Hit', 'Precision']
valid_metric: Hit@10
train_neg_sample_args: ~


eval_args:
  split: {'RS':[0.8,0.1,0.1]}
  group_by: user
  order: RO
  mode: full 
-------------------------------------------------------------------------------------------------------------------------------
#hyper.test

learning_rate choice [0.01,0.005,0.001,0.0005,0.0001]
dropout_prob choice [0.0,0.1,0.2,0.3,0.4,0.5]
mlp_hidden_size choice ['[64,64,64]','[128,128,128]','[256,256,256]','[512,512,512]']

--------------------------------------------------------------------------------------------------------------------------------
運行結果，還為全數運行完，但分數非常低

Trainable parameters: 147647183
Thu 10 Nov 2022 02:34:07 INFO  FLOPs: 57667.0
Thu 10 Nov 2022 02:34:35 INFO  epoch 0 training [time: 28.30s, train loss: 709.3700]
Thu 10 Nov 2022 11:14:57 INFO  epoch 0 evaluating [time: 31221.96s, valid_score: 0.000200]
Thu 10 Nov 2022 11:14:57 INFO  valid result: 
recall@10 : 0.0002    mrr@10 : 0.0001    ndcg@10 : 0.0001    hit@10 : 0.0002    precision@10 : 0.0
Thu 10 Nov 2022 11:15:05 INFO  Saving current: saved/DeepFM-Nov-10-2022_02-34-07.pth
Thu 10 Nov 2022 11:15:32 INFO  epoch 1 training [time: 27.03s, train loss: 467.8100]
Thu 10 Nov 2022 17:44:59 INFO  epoch 1 evaluating [time: 23366.91s, valid_score: 0.000200]
Thu 10 Nov 2022 17:44:59 INFO  valid result: 
recall@10 : 0.0001    mrr@10 : 0.0    ndcg@10 : 0.0001    hit@10 : 0.0002    precision@10 : 0.0
Thu 10 Nov 2022 17:45:08 INFO  Saving current: saved/DeepFM-Nov-10-2022_02-34-07.pth
Thu 10 Nov 2022 17:45:35 INFO  epoch 2 training [time: 26.87s, train loss: 420.0945]
Fri 11 Nov 2022 00:13:26 INFO  epoch 2 evaluating [time: 23271.26s, valid_score: 0.000200]
Fri 11 Nov 2022 00:13:26 INFO  valid result: 
recall@10 : 0.0001    mrr@10 : 0.0001    ndcg@10 : 0.0001    hit@10 : 0.0002    precision@10 : 0.0
Fri 11 Nov 2022 00:13:34 INFO  Saving current: saved/DeepFM-Nov-10-2022_02-34-07.pth
Fri 11 Nov 2022 00:14:01 INFO  epoch 3 training [time: 26.78s, train loss: 378.8792]
Fri 11 Nov 2022 13:40:53 INFO  epoch 3 evaluating [time: 48412.40s, valid_score: 0.000200]
Fri 11 Nov 2022 13:40:53 INFO  valid result: 
recall@10 : 0.0001    mrr@10 : 0.0    ndcg@10 : 0.0001    hit@10 : 0.0002    precision@10 : 0.0
Fri 11 Nov 2022 13:41:04 INFO  Saving current: saved/DeepFM-Nov-10-2022_02-34-07.pth
Fri 11 Nov 2022 13:41:52 INFO  epoch 4 training [time: 48.28s, train loss: 352.4674]
Sat 12 Nov 2022 05:26:19 INFO  epoch 4 evaluating [time: 56666.73s, valid_score: 0.000100]
Sat 12 Nov 2022 05:26:19 INFO  valid result: 
recall@10 : 0.0001    mrr@10 : 0.0    ndcg@10 : 0.0    hit@10 : 0.0001    precision@10 : 0.0
Sat 12 Nov 2022 05:26:45 INFO  epoch 5 training [time: 26.53s, train loss: 328.7948]
Sat 12 Nov 2022 11:49:55 INFO  epoch 5 evaluating [time: 22989.97s, valid_score: 0.000100]
Sat 12 Nov 2022 11:49:55 INFO  valid result: 
recall@10 : 0.0001    mrr@10 : 0.0    ndcg@10 : 0.0    hit@10 : 0.0001    precision@10 : 0.0
Sat 12 Nov 2022 11:50:22 INFO  epoch 6 training [time: 26.50s, train loss: 305.1559]
Sat 12 Nov 2022 18:16:16 INFO  epoch 6 evaluating [time: 23153.93s, valid_score: 0.000200]
Sat 12 Nov 2022 18:16:16 INFO  valid result: 
recall@10 : 0.0001    mrr@10 : 0.0    ndcg@10 : 0.0    hit@10 : 0.0002    precision@10 : 0.0
Sat 12 Nov 2022 18:16:25 INFO  Saving current: saved/DeepFM-Nov-10-2022_02-34-07.pth
Sat 12 Nov 2022 18:16:52 INFO  epoch 7 training [time: 26.80s, train loss: 281.2465]
Sun 13 Nov 2022 00:41:16 INFO  epoch 7 evaluating [time: 23064.06s, valid_score: 0.000100]
Sun 13 Nov 2022 00:41:16 INFO  valid result: 
recall@10 : 0.0001    mrr@10 : 0.0    ndcg@10 : 0.0    hit@10 : 0.0001    precision@10 : 0.0
Sun 13 Nov 2022 00:41:42 INFO  epoch 8 training [time: 26.54s, train loss: 258.6193]
Sun 13 Nov 2022 07:02:46 INFO  epoch 8 evaluating [time: 22864.13s, valid_score: 0.000100]
Sun 13 Nov 2022 07:02:46 INFO  valid result: 
recall@10 : 0.0001    mrr@10 : 0.0    ndcg@10 : 0.0    hit@10 : 0.0001    precision@10 : 0.0
Sun 13 Nov 2022 07:03:13 INFO  epoch 9 training [time: 26.67s, train loss: 238.4493]
Sun 13 Nov 2022 13:29:00 INFO  epoch 9 evaluating [time: 23146.85s, valid_score: 0.000100]
Sun 13 Nov 2022 13:29:00 INFO  valid result: 
recall@10 : 0.0001    mrr@10 : 0.0    ndcg@10 : 0.0    hit@10 : 0.0001    precision@10 : 0.0
Sun 13 Nov 2022 13:29:27 INFO  epoch 10 training [time: 26.77s, train loss: 220.9374]
Sun 13 Nov 2022 19:56:43 INFO  epoch 10 evaluating [time: 23236.00s, valid_score: 0.000100]

**Comment by AoiDragon** [comment]
您好，context-aware模型自身可能并不适用于排序类型的推荐，RecBole在这里只是提供一个评测的接口，但并没有测试过模型具体的表现。

**Comment by emilyjeng** [comment]
您好!謝謝解答，想請問為何context-aware模型不適用於排序?

**Comment by Sherry-XLL** [comment]
您好 @emilyjeng！

我们的 RecBole 框架是支持用 context-aware 模型进行 top-k 排序任务，可以参考 #676，#717，#807，#819，#1086，#1242，#1346 和 #1389 的相关配置。然而，RecBole 支持的 context-aware 模型更多是用于 CTR 点击率预测任务，所以在我们的默认设置中，使用 `AUC` 等 value-based 的评测指标是更符合应用场景与论文的传统设置。

关于模型性能的问题，不同数据集会由于用户物品数量、数据集稀疏度等原因产生较大的差异，超参数的调节也是重要的一环。对于 yelp 数据集本身您也可以通过 k-core 过滤等方式提升数据集的质量，但是总的来说两个数据集的结果是无法直接相比的。

对于 DIEN 模型，它的实现需要利用 timestamp 进行序列数据的处理，具体原因可以参考 #818。因此，DeepFM 等一般的 context-aware 模型并不需要时间戳的属性，但 DIEN 需要，所以会产生您的运行结果。

感谢您对伯乐的关注！
